synapse/lib/storm.py

```diff
@@ -2996,42 +2996,110 @@ async def execStormCmd(self, runt, genr):
 
 class HelpCmd(Cmd):
     '''
-    List available commands and a brief description for each.
+    List available information about Storm and brief descriptions of different items.
+
+    Notes:
+
+        If an item is provided, this can be a string or a function.
 
     Examples:
 
-        // Get all available commands and their brief descriptions.
+        // Get all available commands, libraries, types, and their brief descriptions.
 
         help
 
         // Only get commands which have "model" in the name.
 
         help model
+
+        // Get help about the base Storm library
+
+        help $lib
+
+        // Get detailed help about a specific library function
+
+        help --verbose $lib.
+
+        // Get detailed help about a named Storm type
+
+        help --verbose str
+
+        // Get help about a method from a $node object
+
+        <inbound $node> help $node.tags
+
     '''
     name = 'help'
 
     def getArgParser(self):
         pars = Cmd.getArgParser(self)
-        pars.add_argument('command', nargs='?',
-                          help='Only list commands and their brief description whose name contains the argument.')
+        pars.add_argument('-v', '--verbose', default=False, action='store_true',
+                          help='Display detailed help when available.')
+        pars.add_argument('item', nargs='?',
+                          help='List information about a subset of commands or a specific item.')
         return pars
 
     async def execStormCmd(self, runt, genr):
 
-        if not self.runtsafe:
-            mesg = 'help does not support per-node invocation'
-            raise s_exc.StormRuntimeError(mesg=mesg)
+        node = None
+        async for node, path in genr:
+            await self._runHelp(runt)
+            yield node, path
 
-        async for item in genr:
-            yield item
+        if node is None and self.runtsafe:
+            await self._runHelp(runt)
+
+    async def _runHelp(self, runt: Runtime):
+
+        item = self.opts.item
+
+        if item is not None and \
+                not isinstance(item, str) and \
+                not isinstance(item, s_stormtypes.Lib) and \
+                not callable(item):
```

Comment:
Should we allow passing in a StormType as well so you could do something like `inet:fqdn limit 1 | help $node.props` rather than needing to do `help node:props`?

Reply:
We could support a Prim object, assuming that `self.opts.item` resolved to a Prim.

We have to be careful about that though. In the case of `$node.props` that is a ctor which resolves to a primitive object which we can give real information about.  In cases where the value being passed in is not a prim, we don't have a great path forward. Something like `$lib.regex.flags`, `$lib.regex.flags.i`, or `$lib.debug` would not be able to be resolved; unless the command reached back into the AST to try to decipher what the user was asking for.

Reply:
@Cisphyx I expanded the support to more generic StormType help for registered types.